x/bank/keeper/send.go

```diff
@@ -141,6 +163,11 @@ func (k BaseSendKeeper) InputOutputCoins(ctx sdk.Context, inputs []types.Input,
 			return err
 		}
 
+		// Run send hooks if present.
+		if err := k.Hooks().BeforeSendCoins(ctx, inAddress, inAddress, in.Coins); err != nil {
```

Comment:
Providing the same to and from addresses like this, is not desirable. Same goes for the outputs loop below.

At best, the hook will think nothing is happening, but these calls are also telling lies. This call is telling the hook that `in.Coins` is going to be added to `inAddress`. In the outputs loop, it tells the hook that `out.Coins` is being removed from the `outAddress`. Neither are true.

Reply:
yup first implem was to get the conversation going, probably need to sync the loop from the same index so you have n instead of 2n